static/downloadables/parse_associations.py

```diff
@@ -234,26 +163,23 @@ def parse_txt(filteredFilePath, clumpsObjDict, tableObjDict, snpSet, clumpNumDic
     clumpedVariants= set()
     unmatchedAlleleVariants = set()
 
-    if snpCount == 0:
-        unusedTraitStudy = True
-        return sample_map, clumpedVariants, unmatchedAlleleVariants,unusedTraitStudy, snpCount  
-    else:
-        unusedTraitStudy = False
-
+    pValBetaAnnoValType = "|".join([pValueAnno, betaAnnotation, valueType])
     # iterate through each snp/alleles for this study
+    usedSnps = set()
     for snp in studyLines:
+        usedSnps.add(snp)
         alleles = studyLines[snp]
         if alleles != [] and snp != "":
             # this if statement ensures that the trait/study combo actually exists in the tableObjDict for this snp
             # this is necessary due to excluded snps
-            if trait in tableObjDict['associations'][snp]['traits'] and study in tableObjDict['associations'][snp]['traits'][trait]:
+            if trait in tableObjDict['associations'][snp]['traits'] and study in tableObjDict['associations'][snp]['traits'][trait] and pValBetaAnnoValType in tableObjDict['associations'][snp]['traits'][trait][study]:
                 # grab the corresponding pvalue and risk allele
-                pValue = tableObjDict['associations'][snp]['traits'][trait][study]['pValue']
-                riskAllele = tableObjDict['associations'][snp]['traits'][trait][study]['riskAllele']
+                pValue = tableObjDict['associations'][snp]['traits'][trait][study][pValBetaAnnoValType]['pValue']
+                riskAllele = tableObjDict['associations'][snp]['traits'][trait][study][pValBetaAnnoValType]['riskAllele']
 
+                # not going to do strand flipping for txt files/snp input from user
```

Comment:
this comment feels out of place

Reply:
updated the comment, hopefully provides some clarity